### PR TITLE
[Security] Add new commands for `az security`

### PR DESCRIPTION
**Pull request by ohvano**
**Description**
Adding Microsoft.Security/sqlVulnerabilityAssessments CLI support
New CLI commands:
az security va scans [show/list]
az security va results [show/list]
az security va baseline [show/list/delete/update/set]

**Testing Guide**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by ohvano** [comment]
@shimonar1171

**Comment by yonzhan** [comment]
Security

**Comment by ohvano** [comment]
@shimonar1171
@a7mad-3bbas
@DavidTrigano
@talhers

**Review by DavidTrigano — APPROVED** [review]

File: doc/command_guidelines.md [hunk 1]
```diff
@@ -281,13 +281,9 @@ class FooRuleAddAction(argparse._AppendAction):
...
-            name, metric, operation, value = values.split()
-            return FooRule(
-                name=name,
-                metric=metric,
-                operation=operation,
-                value=value
-            )
+            name, metric, operation, value = values
+            rule = FooRule(name=name, metric=metric, operation=operation, value=value)
+            super(FooRuleAddAction, self).__call__(parser, namespace, rule, option_string)
```
Comment (houk-ms) [inline doc/command_guidelines.md:286]: Maybe we can split the changes on docs to another PR? Seems they have no relationship with the new commands.

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -2977,6 +2977,10 @@ Added Cluster commands and trusted_service_access_enabled parameter for Networkr
 * storage blob generate-sas: User delegation SAS token support with --as-user
 * storage container generate-sas: User delegation SAS token support with --as-user
 
+**Security**
+
+* Added new commands group `az security va` and sub groups `az security va scans` `az security va results` `az security va baseline`
+
```
Comment (houk-ms) [inline src/azure-cli/HISTORY.rst:2983]: We don't need modify this file, the history changes will be extracted from PR titles and be added here automatically when we have a release.

**Review by houk-ms — COMMENTED** [review]
@ohvano, can we provide more details in the PR description about the new commands?

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (ohvano) [reply src/azure-cli/HISTORY.rst]: Deleted

File: doc/command_guidelines.md (hunk 1 above)
Comment (ohvano) [reply doc/command_guidelines.md]: Sure, deleted from this PR. I'll open another.

**Review by houk-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        baseline_update_result = self.cmd('az security va sql baseline update --vm-resource-id {} --workspace-id {} --server-name {} --database-name {} --rule-id {} --latest'
+                                          .format(resource_id, workspace_id, server_name, database_name, rule_id)).get_output_in_json()
+        baseline_show_result = self.cmd('az security va sql baseline show --vm-resource-id {} --workspace-id {} --server-name {} --database-name {} --rule-id {}'
+                                        .format(resource_id, workspace_id, server_name, database_name, rule_id)).get_output_in_json()
+        self.assertEqual(baseline_update_result["properties"], baseline_show_result["properties"])
+
+        # Update baseline for single rule with custom results
+        baseline_input = _get_single_baseline_input(selected_scan_result)
+        baseline_update_result = self.cmd('az security va sql baseline update --vm-resource-id {} --workspace-id {} --server-name {} --database-name {} --rule-id {} {}'
+                                          .format(resource_id, workspace_id, server_name, database_name, rule_id, baseline_input)).get_output_in_json()
```
Comment (shimonar1171) [inline src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py:96]: Suggestion - For each API you should use a function that creates the command (execute the command), for example:

_build_update_baseline_command(resource_id, workspace_id, server_name, database_name, rule_id, baseline_inpu):
return 'az security va sql baseline update --vm-resource-id {} --workspace-id {} --server-name {} --database-name {} --rule-id {} {}'
                                          .format(resource_id, workspace_id, server_name, database_name, rule_id, baseline_input)

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        self.parameter_name = parameter_name
+        self.vm_user = vm_user
+        self.vm_password = vm_password
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.skip_delete = skip_delete
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        template = ('az vm create -l {} -g {} -n {} --admin-username {} --admin-password {}'
+                    ' --image MicrosoftSQLServer:SQL2017-WS2016:Enterprise:latest --size Standard_DS2_v2')
```
Comment (shimonar1171) [inline src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py:36]: const/Configuration

Comment (shimonar1171) [inline src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py:36]: const/Configuration

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        try:
+            return kwargs.get(self.resource_group_parameter_name)
+        except KeyError:
+            template = 'To create a virtual machine a resource group is required. Please add ' \
+                       'decorator @{} in front of this preparer.'
+            raise CliTestError(template.format(ResourceGroupPreparer.__name__,
+                                               self.resource_group_parameter_name))
+
+
+class LogAnalyticsWorkspacePreparer(AbstractPreparer, SingleValueReplacer):
```
Comment (shimonar1171) [inline src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py:55]: Why the test function "_get_log_analytics_workspace_key" in not part of this class

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    @LogAnalyticsWorkspacePreparer()
+    def test_va_sql_scenario(self, resource_group, resource_group_location, sqlvm, laworkspace):
+
+        # Use prepared sql virtual machine and log analytics workspace to setup OMS agent with VA management pack
+        _enable_intelligence_pack(resource_group, laworkspace)
+        la_workspace_id = _get_log_analytics_workspace_id(resource_group, laworkspace)
+        la_workspace_key = _get_log_analytics_workspace_key(resource_group, laworkspace)
+        _set_vm_registry(resource_group, sqlvm)
+        _install_oms_agent_on_vm(self, sqlvm, resource_group, la_workspace_id, la_workspace_key)
+        _restart_monitoring_service(resource_group, sqlvm)
```
Comment (a7mad-3bbas) [inline src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py:30]: add a comment that this was done to force a scan

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+def _enable_intelligence_pack(resource_group, workspace_name):
+    intelligence_pack_name = 'SQLVulnerabilityAssessment'
+    template = 'az monitor log-analytics workspace pack enable -n {} -g {} --workspace-name {}'
+    execute(DummyCli(), template.format(intelligence_pack_name, resource_group, workspace_name))
+
+
+def _set_vm_registry(resource_group, sqlvm):
+    template = 'az vm run-command invoke --command-id {} --name {} -g {} --scripts \'New-Item -ItemType Directory -Force -Path C:\\Users\\admin123\\Desktop\\Va_Logs\' \
+                \'New-Item -ItemType Directory -Force -Path C:\\Users\\admin123\\Desktop\\Setup_Logs\' \
```
[the remark this answers is not in the document]
Comment (ohvano) [reply src/azure-cli/azure/cli/command_modules/security/tests/latest/test_va_sql_scenario.py:127]: For debugging, if needed

File: src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py (hunk 5 above)
Comment (ohvano) [reply src/azure-cli/azure/cli/command_modules/security/tests/latest/preparers.py:55]: This is a preparer class which is part of the CLI engine. These are classes that automate the process of resources creation. You shouldn't call these classes in the middle of the test.

**Comment by msyyc** [comment]
CI failed, please fix it.

**Review by msyyc — APPROVED** [review]

**Comment by dbradish-microsoft** [comment]
@ohvano, I am not seeing MicrosoftDoc/x content for these as security references.  Can you please reach out to me at dbradish@microsoft.com so I can plan for docs content?  Thank you.
